pkg/volume/util/util.go

```diff
@@ -539,6 +579,25 @@ func ChooseZoneForVolume(zones sets.String, pvcName string) string {
 	return zone
 }
 
+// chooseZonesForVolumeIncludingZone is a wrapper around ChooseZonesForVolume that ensures zoneToInclude is chosen
+func chooseZonesForVolumeIncludingZone(zones sets.String, pvcName, zoneToInclude string, numReplicas uint32) (sets.String, error) {
+	if uint32(zones.Len()) < numReplicas {
+		return sets.NewString(), fmt.Errorf("not enough zones found to provision a volume with %d replicas. Need at least %d distinct zones for a volume with %d replicas", numReplicas, numReplicas, numReplicas)
+	}
+	if uint32(zones.Len()) == numReplicas {
+		return zones, nil
+	}
+	if zoneToInclude != "" {
```

Comment:
What if `zoneToInclude` is not in `zones`?

Reply:
If this function assumes `zoneToInclude` is always in `zones`, it'd be helpful to call it out in comments

Reply:
The scheduler will make sure `zoneToInclude` in always in `zones`. Will call out the assumption explicitly.

Reply:
Maybe you are referring to a different function? Both calls to `choozeZonesForVolumeIncludingZone()` pass in non-empty `zones` and they both allow for the possibility of non-empty `zoneToInclude`

Reply:
Sorry - I was indeed referring to a different function. Agree with your comment above - `choozeZonesForVolumeIncludingZone()` does expect `zoneToInclude` to be part of `zones` and I will clarify that in a comment.